File: guides/micronaut-eclipsestore-persitence/java/src/test/java/example/micronaut/FruitRepositoryTest.java
package example.micronaut;

import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;
import org.junit.jupiter.api.function.Executable;
import jakarta.validation.ConstraintViolationException;
import static org.junit.jupiter.api.Assertions.assertThrows;

@MicronautTest(startApplication = false) // <1>
class FruitRepositoryTest {

    @Inject
    FruitRepository fruitRepository;

    @Test
    void methodsValidateParamers() {
        Executable e = () -> fruitRepository.create(new FruitCommand(""));
        assertThrows(ConstraintViolationException.class, e);
    }
}

File: guides/micronaut-microstream-persistence/kotlin/src/test/kotlin/example/micronaut/FruitRepositoryTest.kt
package example.micronaut

import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import jakarta.validation.ConstraintViolationException

@MicronautTest(startApplication = false) // <1>
class FruitRepositoryTest {

    @Inject
    lateinit var fruitRepository: FruitRepository

    @Test
    fun methodsValidateParameters() {
        Assertions.assertThrows(ConstraintViolationException::class.java) {
            fruitRepository.create(FruitCommand(""))
        }
    }
}